benchmark/report.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+"""
+Report the results of a benchmark run recorded by benchmark.py.
+
+Usage:
+    ./report.py PROGRAM0.stats PROGRAM1.stats ...
+"""
+
+import sys
+import numpy as np
+
+def report_file(file_name: str) -> np.array:
+    data_ns = np.genfromtxt(file_name, dtype=int)
+    num_files, num_iterations = data_ns.shape
```

Comment:
This seems to fail when there's just a single row of data (i.e. `num_files == 1`) because numpy probably infers a single-dimensional shape then. 

```
Traceback (most recent call last):
  File "/home/fatho/channable/alfred-margaret/benchmark/./report.py", line 33, in <module>
    report_file(file_name)
  File "/home/fatho/channable/alfred-margaret/benchmark/./report.py", line 15, in report_file
    num_files, num_iterations = data_ns.shape
ValueError: not enough values to unpack (expected 2, got 1)
```

Reply:
Okay, I don't know a lot about numpy but I think it's possible to tell it what shape to expect. I'll update the script.

Reply:
The script works for a single file, it doesn't work for the ` .results` files though. Those contain only a single number per input file (the number of matches). `report.py` is meant to be used with the `.stats` files which contain the timings.

Reply:
I did use it with the `.stats` files. The `num_files` here refers to the number of input files I think. And when there's just a single input file (the `example.txt`) then the stats files contain just a single row, and that's when it breaks.

It's a bit of an edge case :sweat_smile:

Reply:
Ahh thanks, I got it now. Last commit fixes this issue by using a numpy function that allows us to specify the minimum number of dimensions. Outputs are consistent with the old version for `.stats` files I got before.